Service/src/routes/transaction.py

```diff
@@ -1,11 +1,110 @@
-from flask import Blueprint, request, Response, g
+from flask import Blueprint, jsonify, make_response, request, Response, g
 import simplejson as json
 
+from auth import auth
+from .charts import get_history
+from request_validator import validator
+from request_validator.schemas import transaction_schema
 from util.utility import Utility
+from util.error_map import errors
 
 transaction_api = Blueprint('transaction_api', __name__)
 
 
+@transaction_api.route('/api/transactions', methods=['POST'])
+@auth.login_required
+@validator.validate_body(transaction_schema.fields)
+def post_transaction():
+   body = request.get_json()
+   if not auth.portfolio_belongsTo_user(body['portfolioId']):
+      return Response(status=403)
+
+   try:
+      sharePrice = json.loads(get_history(body['ticker'], 'recent'))[0]['price']
+   except TypeError:
+      return make_response(jsonify(UnsupportedTicker=errors['unsupportedTicker']), 400)
+
+   if body['transType'] == 'BUY':
+      return buy(sharePrice, body)
+   else:
+      return sell(sharePrice, body)
```

Comment:
Will this default to sell if transType isn't buy? If so, I don't think it's a good idea to sell shares if someone puts an invalid transType